### PR TITLE
:book: Update documentation for "Finding extensions to install"

### PR DESCRIPTION
**Pull request by michaelryanpeter**
# Description
* https://github.com/operator-framework/operator-controller/issues/693
* Add a _copy_ button for code blocks and support for admonitions
* Add "Catalog queries" and reference documentation directory
* Update and rename "Finding extensions to install"

## Preview links
* References > [Catalog queries](https://deploy-preview-696--olmv1.netlify.app/refs/catalog-queries/)
* Tasks > [Finding extensions to install](https://deploy-preview-696--olmv1.netlify.app/tasks/explore-available-packages/)

## Reviewer Checklist

- [ ] API Go Documentation
- [ ] Tests: Unit Tests (and E2E Tests, if appropriate)
- [ ] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

### CODE REVIEW

File: docs/refs/catalog-queries.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+You can use the `curl` command with `jq` to query catalogs that are installed on your cluster.
+
+``` terminal title="Query syntax"
+$ curl http://localhost:8080/catalogs/operatorhubio/all.json | <query>
+```
+
+## Package queries
+
+Available packages in a catalog
+: `jq -s '.[] | select( .schema == "olm.package")`
```
Comment (everettraven) [inline docs/refs/catalog-queries.md:12]: Nit: For all of the `jq` commands shown here can we use the "code fence" syntax (i.e triple backtick)? This results in a user being able to easily copy the command to their clipboard via a handy dandy button and IMO presents a nicer docs experience when used where possible.

**Review by everettraven — APPROVED** [review]
One minor nit that I don't think should hold this PR. Otherwise looks good to me!

File: docs/refs/catalog-queries.md (hunk 1 above)
Comment (michaelryanpeter) [reply docs/refs/catalog-queries.md:12]: I created a follow up issue: https://github.com/operator-framework/operator-controller/issues/698

Basically, I can't get the code blocks to render in the definition list. I think I might need to move the list to a table, but if I get the code block to work, the table gets stretched really wide. 

In the interest of making the Kubecon deadline, I would prefer to get the rest of the content in place and circle back after. WDYT, @everettraven?

**Review by tmshort — APPROVED** [review]
/lgtm

File: docs/refs/catalog-queries.md (hunk 1 above)
Comment (everettraven) [reply docs/refs/catalog-queries.md:12]: SGTM - I don't think this nit should at all block this from merging!

File: docs/Tasks/explore-available-packages.md [hunk 2]
```diff
@@ -1,135 +1,141 @@
...
-  icon:
-    data: PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiIHN0YW5kYWxvbmU9Im5vIj8+CjwhRE9DVFlQRSBzdmcgUFVCTElDICItLy9XM0MvL0RURCBTVkcgMS4xLy9FTiIgImh0dHA6Ly93d3cudzMub3JnL0dyYXBoaWNzL1NWRy8xLjEvRFREL3N2Z==
-    mediatype: image/svg+xml
-  packageName: argocd-operator
-status: {}
-```
-
-**This CR is most helpful when exploring the versions of a package that are available for installation on cluster, and the upgrade graph of versions** (eg if v0.5.0 of `argocd-operator` is installed on cluster, what is the next upgrade available? The answer is v0.6.0).
\ No newline at end of file
+After you add a catalog of exetensions to your cluster, you must port forward your catalog as a service.
```
Comment (yannizhang2019) [inline docs/Tasks/explore-available-packages.md:1]: typo `exetensions` should be `extensions`

File: docs/Tasks/explore-available-packages.md [hunk 3]
```diff
@@ -1,135 +1,141 @@
...
+Then you can query the catalog by using `curl` commands and the `jq` CLI tool to find extensions to install.
+
+## Prerequisites
+
+* You have added a catalog of extensions, such as [OperatorHub.io](https://operatorhub.io), to your cluster.
+* You have installed the `jq` CLI tool.
+
+## Procedure
+
+1. Port forward the catalog server sevice:
```
Comment (yannizhang2019) [inline docs/Tasks/explore-available-packages.md:11]: `sevice` should be `service`

File: docs/Tasks/explore-available-packages.md [hunk 4]
```diff
@@ -1,135 +1,141 @@
...
+    ``` terminal
+    $ kubectl -n catalogd-system port-forward svc/catalogd-catalogserver 8080:80
+    ```
+
+2. Return a list of all the extensions in a catalog:
+    ``` terminal
+    $ curl http://localhost:8080/catalogs/operatorhubio/all.json | jq -s '.[] | select(.schema == "olm.package") | .name'
+    ```
+
+    ??? success
```
Comment (yannizhang2019) [inline docs/Tasks/explore-available-packages.md:22]: what does `???` provide?

File: docs/Tasks/explore-available-packages.md (hunk 3 above)
Comment (michaelryanpeter) [reply docs/Tasks/explore-available-packages.md:11]: TY! Addressed in https://github.com/operator-framework/operator-controller/pull/728

File: docs/Tasks/explore-available-packages.md (hunk 2 above)
Comment (michaelryanpeter) [reply docs/Tasks/explore-available-packages.md:1]: TY! Addressed in https://github.com/operator-framework/operator-controller/pull/728

File: docs/Tasks/explore-available-packages.md (hunk 4 above)
Comment (michaelryanpeter) [reply docs/Tasks/explore-available-packages.md:22]: Sorry, I had a draft for this comment, but I forgot to submit it. 

The `???` creates a code block with a toggle.

**Pull request merged**